### PR TITLE
Update developer guide (add/delete/swap `Task`)

### PR DESCRIPTION
**Pull request by teoyuqi**
Updated developer guide for functionality of add, delete and swap task.

Also fixed a minor misnaming of variables in code committed earlier for `v1.2`.

### CODE REVIEW

File: docs/diagrams/AddDeleteSwapTaskFeature/AddressBookObjectDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+UniqueModuleList *--> Module1
+Module1 *--> "1" ModuleCode1
+Module1 *--> "1" ModuleTitle1
+Module1 *--> "1" TaskList1 MODEL_COLOR_T5
+
+UniqueModuleList *----> Module2
+Module2 *--> "1" ModuleCode2
+Module2 *--> "1" ModuleTitle2
+Module2 *--> "1" TaskList2 MODEL_COLOR_T5
```
Comment (cheeheng) [inline docs/diagrams/AddDeleteSwapTaskFeature/AddressBookObjectDiagram.puml:35]: Is there a reason why multiplicity is present in the object diagram?

File: docs/diagrams/AddDeleteSwapTaskFeature/AddTaskSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Model --> AddTaskCommand
+deactivate Model
+
+AddTaskCommand --> LogicManager : commandResult
+destroy AddTaskCommand
+
+[<--LogicManager : commandResult
+deactivate LogicManager
+@enduml
```
Comment (cheeheng) [inline docs/diagrams/AddDeleteSwapTaskFeature/AddTaskSequenceDiagram.puml:79]: This sequence diagram might look a bit complicated but I feel that it couldn't be helped due to long method names. Let's discuss. This is not a request to change the sequence diagram.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -156,6 +156,132 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+**Step 11**: The `Module` currently existing in the `Model` is then 
+replaced with this new updated `Module` using the `Model::setModule` method.
+
+The following sequence diagram summarizes what happens when a user executes 
+the `add-task` command:
+
+![AddTaskSequenceDiagram](images/AddDeleteSwapTaskFeature/AddTaskSequenceDiagram.png)
+
+#### Design considerations:
+**Aspect: data structure to store `Task`:**
```
Comment (cheeheng) [inline docs/DeveloperGuide.md:257]:
```suggestion
**Aspect: Data structure to store `Task`:**
```

Sorry for the minor nitpick, but could this capitalisation change be made so as to be consistent with line 270 (new index)?

File: docs/diagrams/AddDeleteSwapTaskFeature/TaskListClassDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        Class Task MODEL_COLOR_T5
+    }
+}
+
+UniqueModuleList *--> "~* all" Module
+
+Module *--> "1" ModuleCode
+Module *--> "1" ModuleTitle
+Module *-right-> "1" TaskList MODEL_COLOR_T5
+
```
Comment (cheeheng) [inline docs/diagrams/AddDeleteSwapTaskFeature/TaskListClassDiagram.puml:24]: Is there a reason why `Link` is left out? I feel that `Link` should be present due to completeness, or that `ModuleCode` and `ModuleTitle` attributes could be left out if the desire is to focus on primarily the `Task` and `TaskList`.

**Review by cheeheng — CHANGES_REQUESTED** [review]
LGTM except for a few minor nitpicks.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -156,6 +156,132 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+  `Module` in Plannit.
+- `SwapTaskCommand` allows the user to swap the order of `Task`s within a 
+  specified `Module` in Plannit.
+
+Given below is an example usage scenario and how the mechanism 
+behaves when a user adds a new `Task`. The behavior for the deleting and 
+swapping of tasks is highly similar.
+
+**Step 1**. The user requests to add a task into a module present in the 
+application by inputting the `add-task` command followed by the `m/` flag to
```
Comment (Tan-Jia-Rong) [inline docs/DeveloperGuide.md:210]:
```
We have implemented the following `Command` classes:
- `AddTaskCommand` allows the user to add a task to a specified `Module` in 
  **Plannit**.
- `DeleteTaskCommand` allows the user to delete a task from a specified 
  `Module` in **Plannit**.
- `SwapTaskCommand` allows the user to swap the order of `Task`s within a 
  specified `Module` in **Plannit**.
```
For consistency
```suggestion
Plannit by inputting the `add-task` command followed by the `m/` flag to 
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -156,6 +156,132 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+**Step 10**: A new `Module` is then created using the copied fields along 
+with the updated `TaskList` field.
+
+**Step 11**: The `Module` currently existing in the `Model` is then 
+replaced with this new updated `Module` using the `Model::setModule` method.
+
+The following sequence diagram summarizes what happens when a user executes 
+the `add-task` command:
+
+![AddTaskSequenceDiagram](images/AddDeleteSwapTaskFeature/AddTaskSequenceDiagram.png)
```
Comment (Tan-Jia-Rong) [inline docs/DeveloperGuide.md:254]: Require a note to indicate plantuml constraints

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -156,6 +156,132 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+**Step 2**: The `LogicManager` calls the `LogManager::execute` method on the 
+user input `String`.
+
+**Step 3**: The `LogicManager::execute` method first parses the user input 
+`String` into a `Command` object using the `AddressBookParser::parseCommand` 
+method.
+
+**Step 4**: The command word, `add-task`, is extracted from the user input and
+a new `AddTaskComamandParser` is instantiated to parse the arguments.
```
[the remark this answers is not in the document]
Comment (teoyuqi) [reply docs/DeveloperGuide.md:226]: Good catch, thank you! :)

File: docs/diagrams/AddDeleteSwapTaskFeature/AddressBookObjectDiagram.puml (hunk 1 above)
Comment (teoyuqi) [reply docs/diagrams/AddDeleteSwapTaskFeature/AddressBookObjectDiagram.puml:35]: ooops ok I will remove it. Thank you for spotting it!

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -156,6 +156,132 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+**Step 5**: The `AddTaskCommandParser::parse` method is then called to 
+parse the arguments. After validating the arguments provided by the user, a 
+new `Task` is instantiated with the provided description.
+
+**Step 6**: An `AddTaskToModuleDescriptor` object is then instantiated to 
+contain the new `Task` and `ModuleCode` of the module to add the `Task` to.
+
+**Step 7**: This `AddTaskToModuleDescriptor` is then used to instantiate an 
+`AddTaskCommand` object that is returned to the `LogicManager`.
```
Comment (ekweirui) [inline docs/DeveloperGuide.md:236]: I feel that we can cut back on some details here, especially step 4. 

One guideline that I adopt when coming up with steps like these is to only include a step when I notice a need to include an object diagram to showcase a useful change after a particular scenario happened. 

For example in the scenario here, I would skimp past the details in step 3, 4 that are related to parsing, and focus on steps like 5, and 6 where I think object diagrams can be very helpful in helping to create a narrative here.

However, this is just a suggestion as this is just my personal style, but I do hope you find it useful~! 😄

**Review by ekweirui — DISMISSED** [review]
LGTM except for a minor nitpick.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (teoyuqi) [reply docs/DeveloperGuide.md:257]: No worries! Thank you for pointing that out 😃

File: docs/diagrams/AddDeleteSwapTaskFeature/TaskListClassDiagram.puml (hunk 4 above)
Comment (teoyuqi) [reply docs/diagrams/AddDeleteSwapTaskFeature/TaskListClassDiagram.puml:24]: Good catch, I will add `Link` into the diagram.

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (teoyuqi) [reply docs/DeveloperGuide.md:345]: Thank you for pointing out!

**Review by Tan-Jia-Rong — APPROVED** [review]
LGTM

**Review by cheeheng — APPROVED** [review]
LGTM

**Pull request merged**